cmd/kubelet/app/options/container_runtime.go

```diff
@@ -51,7 +51,6 @@ func NewContainerRuntimeOptions() *config.ContainerRuntimeOptions {
 		DockershimRootDirectory:    "/var/lib/dockershim",
 		PodSandboxImage:            defaultPodSandboxImage,
 		ImagePullProgressDeadline:  metav1.Duration{Duration: 1 * time.Minute},
-		ExperimentalDockershim:     false,
```

Comment:
presumably this was still alpha.

I may have missed this (apologies), but is there a replacement available for this?

Reply:
Hmmm... in the commit message, I wrote that I didn't think anyone was using this:

```
I'm fairly confident, though need to double check, that no one is
actually using this experimental dockershim behavior. If they are, I
think we will want to find a new way to support it (that doesn't require
using the Kubelet only to launch Dockershim).
```

Looking at the code, I'm seeing this setting defaults to false and is also marked as "use at your own risk" (see the code [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/config/flags.go#L92)).

@dims I believe you added this flag - can you confirm that its safe to delete? Apologies if I already asked you this question :)

Reply:
still alpha = we can delete if we want :sweat_smile: 

I may be mis-remembering this flag, but we probably want a dockershim-as-cri thing somewhere to help people migrate off of the built-in one?

unless we actually predict full migration off of dockerd, which I think might be looking less likely in the immediate future?

Reply:
once this lands i will push for a full migration off dockershim. if someone wants to write a CRI for dockerd, they  will be welcome to do so.

Reply:
relevant then: https://github.com/kubernetes/enhancements/pull/1717
(dockershim is intertwined with not GCing the pod sandbox image currently)

Reply:
The dockershim was done as part of CRI work by @Random-Liu if I recall.  The `experimental-dockershim-*` flags were also often used when running containerized kubelet back in 1.11 timeframe with testing.  In general, what we do when this flag is started versus what we do when the runtime chosen is docker is primarily differentiated only by how the streaming server is started.  Either way, +1 on its removal.

Reply:
@BenTheElder the pod sandbox image option works even when not using the docker container runtime.  ideally, we could query the CRI to know the image it uses as well so we would not need to specify the flag at the kubelet level.

Reply:
It's true, and I do think that would be neat but:
- Theoretically I don't see why I couldn't implement CRI with a "pod" with some backend that doesn't use such an image
- There are other images that should not be GCed, it's unfortunate that we cannot specify this today.

Not a blocker for this, just though dims and matt might be interested :-)

Reply:
well it landed and cri-tools CI test is now predictably broken...

Reply:
cc @dims 

Mind linking the failing test Mike? Thanks!

Reply:
Not sure how I missed your request.. @mattjmcnaughton https://github.com/kubernetes-sigs/cri-tools/pull/611

Reply:
Thanks Mike!